src/aio.pxi

```diff
@@ -0,0 +1,99 @@
+'''
+aio.pxi
+
+asyncio/Trio compatibility layer for pyfuse3
+
+Copyright © 2018 JustAnotherArchivist, Nikolaus Rath <Nikolaus.org>
+
+This file is part of pyfuse3. This work may be distributed under
+the terms of the GNU LGPL.
+'''
+
```

Comment:
This is pretty neat code, I'm glad the differences can be encapsulated away so nicely!

What I had in mind was a slightly different usage, where `aio` is a replacement for Trio, not an intermediate layer. In other words, the usage would not be

```
import aio
aio._set_aio(CHOICE)
lock = aio.Lock() # etc
```

```
if CHOICE:
    import trio
else:
   import trio_asyncio as trio  # formerly aio
lock = aio.Lock() # etc
```

Do you have any thoughts on that? It seems nicer to me because that way `trio_asyncio` doesn't need to make all the runtime switching between trio and asyncio, and elsewhere in the code this can go away as well (i.e., no more `if` to switch between asyncio.sleep and trio.sleep).

Reply:
I think this will be easy. Even when using `trio.hazmat`, the import is just `import trio`. So in the compatibility layer could just do:

```
import argparse

def function_in_hazmat(bla):
  pass
hazmat = argparse.Namespace()
hazmat.function_in_hazmat = function_in_hazmat
```

As for the name of that module, I really don't care. I just wrote the first thing that came to mind :-).